### PR TITLE
#404 Add Sonarcloud to Github actions & exclude files

### PR DESCRIPTION
**Pull request by JeremiahTee**
This PR adds Sonarcloud to our Github actions so that the coverage is ran at commit level in open PRs. Sonarcloud was suggested by our Prof during Release #2 feedback
This also excludes testData.ts as it is flagged by Sonarqube by having lots of duplications -> we concluded they are false positives

**Notes**
- This must be merged into master for it to work

### CODE REVIEW

**Comment by JeremiahTee** [comment]
Sonar fails as it's not able to find the project settings, will look into it

**Comment by JeremiahTee** [comment]
Fixed build.yml, Sonarcloud is now running properly. Test coverage can be ignored as it is being taken care of jest (which is already set up in the CI pipeline)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,4 +33,16 @@ jobs:
...
+          MAILGUN_API_KEY: ${{ secrets.MAILGUN_API_KEY }}
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
```
Comment (GD-Dheer) [inline .github/workflows/build.yml:45]: Nice

**Review by GD-Dheer — APPROVED** [review]
You should merge this into dev then we can have an additional PR into master since this is adding changes independent of sonar into master

**Comment by JeremiahTee** [comment]
> You should merge this into dev then we can have an additional PR into master since this is adding changes independent of sonar into master

Gotcha, I switched base to develop

**Review by Vatika1 — APPROVED** [review]
👏🏼👍🏼

**Pull request merged**